crates/api_common/src/utils.rs

```diff
@@ -199,16 +214,14 @@ pub async fn check_community_mod_action(
   Ok(())
 }
 
-pub async fn check_community_mod_action_opt(
+/// Check that a person is either a mod of any community, or an admin
+///
+/// Should only be used for read operations
+pub async fn check_community_mod_of_any_or_admin_action(
   local_user_view: &LocalUserView,
-  community_id: Option<CommunityId>,
   pool: &mut DbPool<'_>,
 ) -> LemmyResult<()> {
-  if let Some(community_id) = community_id {
-    check_community_mod_action(&local_user_view.person, community_id, false, pool).await?;
-  } else {
-    is_admin(local_user_view)?;
```

Comment:
Here was the issue: if a community wasn't given (which it often isn't in the case of fetching all of a mods reports, or the report count), it would only check for admin status.

I changed this by adding a function which checks if you're the mod of *any* community, or an admin.